pennylane/devices/task_qubit.py

```diff
@@ -0,0 +1,372 @@
+    As an example, we can modify the above example to adjust the supplied weights for circuit execution, and gather the
+    results altogether when complete.
+
+    >>> func = qml.taskify(my_workflow, futures=True)
+    >>> futures = [func(i*np.array([1.0,2.0,3.0]), "default.qubit", "autograd", "backprop") for i in range(5)]
+    >>> futures
+    [<Future: finished, type: numpy.ndarray, key: my_workflow-238c01c1b1b88ef140d2a0404a226f9e>,
+     <Future: finished, type: numpy.ndarray, key: my_workflow-276bf61f8de5488ec4e05ae93e3d0b85>,
+     <Future: finished, type: numpy.ndarray, key: my_workflow-ccbf9977e25ec480b726b93c88cfc9e3>,
+     <Future: finished, type: numpy.ndarray, key: my_workflow-e644cd634c006efc6fc7dd14837ed639>,
+     <Future: finished, type: numpy.ndarray, key: my_workflow-ce39fdf431490ef17c836800b4a7318b>]
+    >>> qml.untaskify(futures)()
+    [array([[0., 0., 0.],
+        [0., 0., 0.],
+        [0., 0., 0.]]),
+     array([[-3.74614396e-01,  2.62791617e-01,  1.71438687e-02],
+            [ 1.11022302e-16,  9.00197630e-01,  5.87266449e-02],
+            [-9.09297427e-01, -1.38777878e-16, -3.46944695e-17]]),
+     array([[-4.74976196e-01,  1.25841537e-01, -3.16289455e-02],
+            [ 1.66533454e-16,  7.26659269e-01, -1.82638158e-01],
+            [ 7.56802495e-01,  9.71445147e-17,  6.93889390e-18]]),
+     array([[-2.44443912e-01, -2.49513914e-01, -3.87823537e-01],
+            [ 2.77555756e-17, -2.54583916e-01, -3.95703924e-01],
+            [ 2.79415498e-01,  0.00000000e+00,  0.00000000e+00]]),
+     array([[ 1.21474177e-01, -3.56699848e-01, -3.33559948e-02],
+            [ 2.77555756e-17, -8.34873873e-01, -7.80713777e-02],
+            [-9.89358247e-01, -2.22044605e-16,  1.38777878e-17]])]
+
+    Args:
+        wires (int): The number of wires to initialize the device with.
+        backend (None, str): Indicates the PennyLane device type to use for offloading
+            computation tasks. This can be `default.qubit`, one of its subclasses or
+            `lightning.qubit`. The TensorFlow, PyTorch or JAX interfaces are the preferred types
+            for gradient computations. This is due to existing support in Dask for
+            TF and Torch datatypes, and natural support for JAX via numpy.
+        future (None, bool): Indicates whether the internal circuit evaluation returns a future
+            to a result. This allows building of dependent workflows, but currently only works with
+            explicit calls to `device.batch_execute` with a PennyLane native device type such as
+            (`default.qubit`, `lightning.qubit`).
+        gen_report (bool, str): Indicates whether the backend task-scheduler will generate a performance report based on the tasks that were run.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+    # pylint: disable=no-self-argument
+
+    operations = DefaultQubit.operations
+    observables = DefaultQubit.observables
```

Comment:
I agree. I think right now it needs to be a device, but I reckon it can become a QNode subclass after some refactoring of other sections. Tom and I discussed some future facing plans for this, and the plan is to have this available, but not explicitly declared (ie dev & research use only).
Long term, decoupling the circuit description and device memory allocation within a qnode (ie some type of MVC design) makes sense here, as we can opt for any executor that follows the `concurrent.futures` API.

Reply:
Also, just on the topic of the operations/observables, these aren't required anymore, as they are pulled in from the parent-class.